plugins/out_stackdriver/stackdriver.c

```diff
@@ -567,7 +880,35 @@ static int stackdriver_format(struct flb_config *config,
         /* jsonPayload */
         msgpack_pack_str(&mp_pck, 11);
         msgpack_pack_str_body(&mp_pck, "jsonPayload", 11);
-        msgpack_pack_object(&mp_pck, *obj);
+
+        if (k8s_resource_type) {
+            map_size = obj->via.map.size;
+            new_map_size = map_size;
+            msgpack_object_kv *kv = NULL; 
+
+            for (i = 0; i < map_size; ++i) {
+                kv = &obj->via.map.ptr[i];
+                if (strncmp(kv->key.via.str.ptr, LOCAL_RESOURCE_ID_KEY, len) == 0) {
```

Comment:
to optimize you can check for the key length first and then do the strncmp()

Reply:
updated